### PR TITLE
tasks/rsync: This code has new task rsync

### PR DESCRIPTION
**Pull request by ramkiperiy**
1. rsync can be used in parallel with other tasks in cephFS testing.
    2. This runs in parallel with workunit, i.e it can sync IO data from workunit to ceph mount
    3. Snapshots can be used to rsync data, snapshot created on workunit directory
    4. Rsync uses observer thread, which checks event state from workunit task.

cherry-picked task/workunit changes from John's PR.

### CODE REVIEW

**Comment by jcsp** [comment]
So I've just tried running this with some .yaml to invoke it, and I'm getting "Could not find task rsync" errors (http://pulpito.ceph.com/jspray-2017-01-10_00:05:12-fs:basic-master-distro-basic-smithi/)

Possibly that's just an indication that the import of the rsync module was failing for some reason?

The commit I added is here:
https://github.com/jcsp/ceph/commit/707bcf057c479848bfaf7f80bed7a5178beea0cf

I was scheduling this with
```
teuthology-suite --suite fs/basic -m smithi -p 99 -c master --ceph-repo=https://github.com/ceph/ceph.git --newest 10 --suite-repo=https://github.com/jcsp/ceph.git --suite-branch task-rsync -k distro
```

**Comment by ramkiperiy** [comment]
John,

Did i committed the patch in wrong location ?, because of that it's unable to find the task. 

Regards,
Ramakrishnan

**Comment by ramkiperiy** [comment]
John,

Did i committed the patch in wrong location ?, because of that it's unable
to find the task.

Regards,
Ramakrishnan

File: tasks/rsync.py [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+
+'''
+    parameters:
+
+    filesize: Optional Param. Unit in MB, default file size 2MB
+              Size of the data file before creating snapshot.
+              Not of use if workunit task running in parallel.
+
+    runtime: Required Param for using rsync task sequentially. unit in seconds. default 0 sec.
+             Don't use this param during parallel execution. if used then finally block will be delayed
```
Comment (amitkumar50) [inline tasks/rsync.py]: @ramkiperiy (NIT) Looks fullstop missed at end.

File: tasks/rsync.py [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+
+        # Create destination directory
+        self.my_mnt.run_shell(["mkdir", "rsyncdir"])
+
+        if self.snap_enable:
+            # Enable snapshots
+            self.fs.mon_manager.raw_cluster_cmd("mds", "set", "allow_new_snaps", "true", "--yes-i-really-mean-it")
+
+        if not self.work_unit:
+            # Create a data directory, sub directory and rsync directory
```
Comment (amitkumar50) [inline tasks/rsync.py]: Hope we have not missed creating 3 directories here.
Please ignore if done

**Comment by liewegas** [comment]
@batrick is this still something we weant?

**Comment by batrick** [comment]
Tracker for this PR: https://tracker.ceph.com/issues/17434

**Pull request closed without merging**